pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -143,3 +160,78 @@ func manageOpenShiftControllerManagerPublicConfigMap_v311_00_to_latest(client co
 
 	return resourceapply.ApplyConfigMap(client, configMap)
 }
+
+func manageBuildAdditionalCAConfigMap(client coreclientv1.ConfigMapsGetter, configClient configclientv1.BuildsGetter, operatorConfig *v1alpha1.OpenShiftControllerManagerOperatorConfig) (string, bool, error) {
+	caMap := resourceread.ReadConfigMapV1OrDie(v311_00_assets.MustAsset("v3.11.0/openshift-controller-manager/build-additional-ca-cm.yaml"))
+	buildConfig, err := configClient.Builds().Get("cluster", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return "", false, err
+	}
+	if buildConfig == nil {
+		caMap.Data = nil
+		_, modified, err := resourceapply.ApplyConfigMap(client, caMap)
+		operatorConfig.Status.AdditionalTrustedCA = nil
+		return "", modified, err
+	}
+
+	caData, err := mergeCAConfigMap(client, buildConfig.Spec.AdditionalTrustedCA)
+	if err != nil {
+		return "", false, err
+	}
+	if len(caData) == 0 && operatorConfig.Status.AdditionalTrustedCA != nil {
+		caMap.Data = nil
+		_, modified, err := resourceapply.ApplyConfigMap(client, caMap)
+		operatorConfig.Status.AdditionalTrustedCA = nil
+		return "", modified, err
+	}
+	h := sha1.New()
+	h.Write(caData)
+	caHash := fmt.Sprintf("%x", h.Sum(nil))
+	if operatorConfig.Status.AdditionalTrustedCA != nil &&
+		operatorConfig.Status.AdditionalTrustedCA.SHA1Hash == caHash {
+		return caHash, false, nil
+	}
+	operatorConfig.Status.AdditionalTrustedCA = &v1alpha1.AdditionalTrustedCA{
+		SHA1Hash:      caHash,
+		ConfigMapName: buildConfig.Spec.AdditionalTrustedCA.Name,
+	}
+	caMap.Data["additional-ca.crt"] = string(caData)
+	_, modified, err := resourceapply.ApplyConfigMap(client, caMap)
+	return caHash, modified, err
+}
+
+// mergeCAConfigMap merges the CA content within the provided ConfigMap. Returns the merged CA bundle bytes.
+//
+// If the ConfigMap contains invalid PEM-encoded data, an error is thrown.
+func mergeCAConfigMap(client coreclientv1.ConfigMapsGetter, cmRef configv1.ConfigMapReference) ([]byte, error) {
+	if len(cmRef.Name) == 0 || len(cmRef.Namespace) == 0 {
+		return nil, nil
+	}
+	configMap, err := client.ConfigMaps(cmRef.Namespace).Get(cmRef.Name, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+	if configMap == nil {
+		return nil, nil
+	}
+	return mergePEMData(configMap.Data)
+}
+
+func mergePEMData(data map[string]string) ([]byte, error) {
+	pemData := &bytes.Buffer{}
+	for key, certData := range data {
+		data := []byte(certData)
+		for len(data) > 0 {
+			block, rest := pem.Decode(data)
+			if block == nil && len(rest) > 0 {
+				return nil, fmt.Errorf("map key %s contains invalid PEM data: %s", key, string(rest))
+			}
+			err := pem.Encode(pemData, block)
+			if err != nil {
+				return nil, err
+			}
+			data = rest
+		}
+	}
```

Comment:
mergePEMData ensures that the provided map contains valid PEM-encoded data for every key. This code may live on even if we are not merging CAs, as IMO the operator should validate that the provided certificates have valid encoding.